structuretoolkit/build/mesh.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import warnings
+from ase.atoms import Atoms
+
+
+def create_mesh(
+    structure: Atoms,
+    n_mesh: int|list = 10,
+    density: float|None = None,
+    endpoint: bool = False
+):
+    """
+    Create a mesh based on a structure
+
+    Args:
+        structure (ase.atoms.Atoms): ASE Atoms
```

Comment:
Would-be-nice: to overload this argument to something like (np.ndarray | ase.atoms.Atoms), and maybe to re-name it `cell`. This function is in `structuretoolkit`, so I won't begrudge you calling it `structure` and making that the priority input, but this is actually a super generic function and should work on any 3x3 cell (maybe with some verification that the cell vectors follow some right hand rule or something, I haven't thought too deeply). 

Anyhow, a sort of multiple-dispatch approach for overloading the input types might be friendly to users.

Reply:
That's actually a good idea. I'm gonna go for it.